File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        int n = s.length();
        int m = spaces.length;
        StringBuilder st = new StringBuilder();
        int i=0, j=0;
        while(i<n && j<m) {
            if(i != spaces[j]) {
                st.append(s.charAt(i));
                i++;
            }
            
            else {
                st.append(' ');
                j++;
            }
        }
        
        while(i<n) {
            st.append(s.charAt(i));
            i++;
        }
        
        return st.toString();
    }
}

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int n = s.size();
        int m = spaces.size();
        int i=0, j=0;
        string st = "";
        while(i<n && j<m) {
            if(i != spaces[j]) {
                st += s[i];
                i++;
            }
            
            else if(i == spaces[j]) {
                st += ' ';
                //st += s[i];
                j++;
            }
        }
        
        while(i<n) {
            st += s[i];
            i++;
        }
        return st;
    }
};